tests/unit/services/date.spec.js

```diff
@@ -0,0 +1,138 @@
+import {
+  resetDateTime,
+  getNDaysFromDate,
+  convertMsToDays,
+  getDaysDateRange,
+  getIntlDateString,
+  generateRandomIndexes,
+  generateDateCycles,
+} from '@/services/date';
+
+
+describe('Date service', () => {
+  const year = 1988;
+  // don't forget that months are 0 based
+  const month = 1;
+  const date = 28;
+  const hour = 7;
+  const minute = 30;
+  const second = 30;
+  const miliSecond = 30;
+  const fullDate = new Date(year, month, date, hour, minute, second, miliSecond);
+
+  it('resetDateTime returns the new date with time set to 00:00:00:00', () => {
+    const resetDate = resetDateTime(fullDate);
+    expect(resetDate.getFullYear()).toBe(year);
+    expect(resetDate.getMonth()).toBe(month);
+    expect(resetDate.getDate()).toBe(date);
+    expect(resetDate.getHours()).toBe(0);
+    expect(resetDate.getMinutes()).toBe(0);
+    expect(resetDate.getSeconds()).toBe(0);
+    expect(resetDate.getMilliseconds()).toBe(0);
+  });
+
+  it(`getNDaysFromDate adds N number of dates to selected date,
+      resets the time to 00:00:00:00 and returns new date`, () => {
+    // we pass in 28.02.1988
+    // when we add 0 days, we expect to keep the same date
+    // and time is set to 0
+    let addedDayes = 0;
+    const sameDate = getNDaysFromDate(fullDate, addedDayes);
+    expect(sameDate.getFullYear()).toBe(year);
+    expect(sameDate.getMonth()).toBe(month);
+    expect(sameDate.getDate()).toBe(date);
+    expect(sameDate.getHours()).toBe(0);
+    expect(sameDate.getMinutes()).toBe(0);
+    expect(sameDate.getSeconds()).toBe(0);
+    expect(sameDate.getMilliseconds()).toBe(0);
+
+    // expect to get 04.03.1988 back, after adding 5 days
+    addedDayes = 5;
+    const resetDate = getNDaysFromDate(fullDate, addedDayes);
+    expect(resetDate.getFullYear()).toBe(year);
+    expect(resetDate.getMonth()).toBe(2);
+    expect(resetDate.getDate()).toBe(4);
+    expect(resetDate.getHours()).toBe(0);
+    expect(resetDate.getMinutes()).toBe(0);
+    expect(resetDate.getSeconds()).toBe(0);
+    expect(resetDate.getMilliseconds()).toBe(0);
+
+    // expect to get 27.02.1988 back, after deducting a day
+    addedDayes = -1;
+    const yesterday = getNDaysFromDate(fullDate, addedDayes);
+    expect(yesterday.getFullYear()).toBe(year);
+    expect(yesterday.getMonth()).toBe(month);
+    expect(yesterday.getDate()).toBe(date - 1);
+    expect(yesterday.getHours()).toBe(0);
+    expect(yesterday.getMinutes()).toBe(0);
+    expect(yesterday.getSeconds()).toBe(0);
+    expect(yesterday.getMilliseconds()).toBe(0);
+  });
```

Comment:
You could have a `describe` block in here and test those stuffs separately  as this test is quite extensive and the test title is big as well

Reply:
Done. :+1:  Does it look more readable?